src/Juvix/Compiler/Backend/Markdown/Translation/FromTyped/Source.hs

```diff
@@ -0,0 +1,181 @@
+module Juvix.Compiler.Backend.Markdown.Translation.FromTyped.Source where
+
+import Commonmark qualified as MK
+import Data.Text qualified as Text
+import Data.Text.IO qualified as Text
+import Data.Text.Lazy (toStrict)
+import Juvix.Compiler.Backend.Html.Data.Options qualified as HtmlRender
+import Juvix.Compiler.Backend.Html.Translation.FromTyped.Source qualified as HtmlRender
+import Juvix.Compiler.Backend.Markdown.Data.Types
+import Juvix.Compiler.Concrete.Data.ScopedName qualified as S
+import Juvix.Compiler.Concrete.Language qualified as Concrete
+import Juvix.Compiler.Concrete.Pretty qualified as Concrete
+import Juvix.Extra.Assets (writeAssets)
+import Juvix.Prelude
+import Text.Blaze.Html
+import Text.Blaze.Html.Renderer.Text qualified as Html
+import Text.Blaze.Html5 as Html hiding (map)
+import Text.Blaze.Html5.Attributes qualified as Attr
+
+data ProcessJuvixBlocksArgs = ProcessJuvixBlocksArgs
+  { _processJuvixBlocksArgsConcreteOpts :: Concrete.Options,
+    _processJuvixBlocksArgsUrlPrefix :: Text,
+    _processJuvixBlocksArgsIdPrefix :: Text,
+    _processJuvixBlocksArgsNoPath :: Bool,
+    _processJuvixBlocksArgsComments :: Comments,
+    _processJuvixBlocksArgsOutputDir :: Path Abs Dir,
+    _processJuvixBlockArgsWriteAssets :: Bool,
+    _processJuvixBlockArgsStdout :: Bool,
+    _processJuvixBlocksArgsModule :: Concrete.Module 'Concrete.Scoped 'Concrete.ModuleTop
+  }
+
+data ProcessingState = ProcessingState
+  { _processingStateMk :: Mk,
+    _processingStateFlag :: Bool,
+    _processingStateSep :: [Int],
+    _processingStateStmts :: [Concrete.Statement 'Concrete.Scoped]
+  }
+
+makeLenses ''ProcessJuvixBlocksArgs
+makeLenses ''ProcessingState
+
+genMarkdownWithJuvixHtml ::
```

Comment:
Suggestion:

Make this function return something of type `Mk`, making it pure, and do all the IO operations (writing files, creating output directory, etc.) in the `Commands.Markdown` module.

This would allow you to more easily write tests for the `genMarkdownWithJuvixHtml` function.